Source/GR32_Math.pas

```diff
@@ -1224,13 +1224,26 @@ function Sign(Value: Integer): Integer;
 {$IFDEF FPC} assembler; nostackframe; {$ENDIF}
 asm
 {$IFDEF TARGET_x64}
-        MOV     EAX, Value
-{$ENDIF}
+  {$IFDEF MSWINDOWS}
+        XOR     EDX, EDX
+        TEST    ECX, ECX
+        SETG    DL
+        SAR     ECX, 31
+        LEA     EAX, [EDX + ECX]
+  {$ELSE}

```

Comment:
You need to select the code block based on some criteria instead of just assuming that everything not Windows can use that calling convention.
I assume that FPC has a symbol defined for your target that can be used.

So somthing like:
```Delphi
{$if defined(Windows)}
        XOR     EDX, EDX
        TEST    ECX, ECX
        [...]
{$elseif defined(YourPlatform)}
        XOR     EDX, EDX
        TEST    EDI, EDI
        [...]
{ifend}
```

Reply:
As far as I know, all non-Windows platforms use the "System V ABI", whether Linux or MacOSX etc; only Microsoft chose to do something different.

Selecting the code block after checking to see if it's Linux or Darwin... I'm bound to miss a platform!  Or do you insist it be done this way?

Reply:
If necessary, a compilation error can be forced with {$ERROR ... } if an architecture/platform isn't fully defined, or just simply fall back to pure Pascal.  At a future date it might be plausible to separate the assembly implementations into separate .inc files.